.github/workflows/pre-commit.yml

```diff
@@ -0,0 +1,43 @@
+name: Dependabot on pre-commit hooks
+
+concurrency:
+  cancel-in-progress: true
+  group: ${{ github.workflow }}-${{ github.ref_name }}
+
+on:
+  schedule:
+    - cron: 0 0 * * 0
+
+jobs:
+  update:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3.5.0
+      - uses: actions/setup-python@v4.5.0
+        with:
+          cache: pip
+          check-latest: true
+          python-version: 3.11
+      - env:
+          PIP_UPGRADE: True
+          PIP_UPGRADE_STRATEGY: eager
```

Comment:
It bothers me a bit and I don't have a clear idea yet. At work we package our application in a .deb so all our dependencies are locked to the precise version of all the libs on Debian. Here I guess one would pipx-install incipyt so it wouldn't use the libraries of the system but rather download and install in a dedicated venv all the libs needed by the application. Then it is a good idea to keep up-to-date with all the libraries. Is that right?